### PR TITLE
Atomic stripe

### PR DESCRIPTION
**Pull request by mattkrick**
Who would've thought that teams would have handfuls of people joining at the _exact_ same second.
If you change quantity twice in 1 second in stripe, it merges them into 1, which causes ugly invoices. 

PS don't mind the merge conflict, it's taken care of in the following PR

TEST
- [ ] create a new pro organization
- [ ] in 2 different browsers, join that paid organization at the same time (e.g. go to the invitation link, fill out a new username/password combo, then hit enter on both quickly)
- [ ] go to the upcoming invoice & view the added members, it works
- [ ] remove members, that works, too

### CODE REVIEW

**Comment by mattkrick** [comment]
@ackernaut ready for review

File: packages/server/graphql/intranetSchema/mutations/stripeUpdateInvoiceItem.ts [hunk 1]
```diff
@@ -1,22 +1,76 @@
...
-    const {quantity} = curHook
-    // if 2 hooks occurred at the same second, try grabbing the one with the accurate quantity
-    if (quantity === hookQuantity) return curHook
+  const firstHook = possibleHooks[possibleHooks.length - 1]
+  const {id: hookId} = firstHook
+  sendToSentry(new Error('Imperfect invoice item hook selected'), {tags: {hookId}})
+  return firstHook
+}
+
+const tagInvoiceItemWithHook = async (invoiceItem: Stripe.invoiceItems.InvoiceItem) => {
```
Comment (mattkrick) [inline packages/server/graphql/intranetSchema/mutations/stripeUpdateInvoiceItem.ts]: @jordanh please review for atomicity. this webhook may get called by stripe multiple times per second

File: packages/server/utils/RedisLock.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import shortid from 'shortid'
+import getRedis from './getRedis'
+
+export default class RedisLock {
```
Comment (mattkrick) [inline packages/server/utils/RedisLock.ts]: @jordanh please review for atomicity. this is how we lock in the app

**Comment by ackernaut** [comment]
### Test cycle no. 1
- [ ] create a new pro organization
- [ ] in 2 different browsers, join that paid organization at the same time (e.g. go to the invitation link, fill out a new username/password combo, then hit enter on both quickly)
- [ ] go to the upcoming invoice & view the added members, it works
- [ ] remove members, that works, too

**Comment by ackernaut** [comment]
Okay, the steps were completed above. The invoice changed in this order:
- $6 (1 user)
- $30 (2 users added)
- $18 (1 of those 2 users removed)

**Pull request merged**